lib/fiddle/struct.rb

```diff
@@ -20,6 +20,33 @@ def CUnion.entity_class
     end
   end
 
+  # Wrapper for arrays within a struct
+  class StructArray < Array
+    include ValueUtil
+
+    def initialize(ptr, type, initial_values)
+      @ptr = ptr
+      @type = type
+      @align = PackInfo::ALIGN_MAP[type]
+      @size = Fiddle::PackInfo::SIZE_MAP[type]
+      @pack_format = Fiddle::PackInfo::PACK_MAP[type]
+      super(initial_values.collect { |v| unsigned_value(v, type) })
+    end
+
+    def to_ptr
+      @ptr
+    end
+
+    def []=(index, value)
+      if index < 0 || index >= size
+        raise RangeError, 'index %d outside of array bounds 0...%d' % [index, size]
+      end
+
+      to_ptr[index * @align, @size] = [value].pack(@pack_format)
```

Comment:
Why `@align`, shouldn't be `@size`?

Reply:
I agree this was wrong. I fixed the error, but I was unable to make it fail in a test case.

The test did not fail because, on my system, `@align == @size` for all primitive types. Therefore the result of using `@align` yields a correct value on my system, but could fail on other architectures. I modified the test case to attempt to write at several array indices, and then check that the underlying memory can be correctly unpacked. I think the modified test case covers the issue better than the original, even though I did not experience a failure, so I committed it.